w2w/w2w.py

```diff
@@ -948,6 +954,25 @@ def add_urb_params_to_wrf(
 
     return nbui_max
 
+def calc_distance_coord(
+        lat1,lon1,lat2,lon2
+):
+
+    '''Calculate distance using coordinates
+       This uses the spherical law of cosines
+    '''
+
+    earth_radius=6371000 #Earth radius in m
+    lat1r = lat1*np.pi/180.
+    lat2r = lat2*np.pi/180.
+    lon1r = lon1*np.pi/180.
+    lon2r = lon2*np.pi/180.
+
+    d = np.arccos(np.sin(lat1r)*np.sin(lat2r) + np.cos(lat1r)*np.cos(lat2r)*np.cos(lon2r-lon1r))*earth_radius
+
+    return d
+
+
```

Comment:
Hey thanks for the implementation, since we are on our way to build up test coverage it would be nice if you could add a test in `tests/w2w_test.py` for this function, looking something like this:

```python
import pytest

@pytest.mark.parametrize(
    ('coords', 'expected'),
    (
        pytest.param((1.2, 3.4, 5.6, 7.8), 9.0, id='nothern hemisphere'),
        pytest.param((5.2, 8.4, 5.6, 7.8), 9.0, id='southern hemisphere'),
        pytest.param((1.2, 3.4, 5.6, 7.8), 9.0, id='across equator'),
    ),
)
def test_calc_distance_coord(coords, expected):
    assert calc_distance_coord(*coords) == pytest.approx(expected)
```
I am not sure if my suggested cases (numbers are fictional) really apply for the function, but I can think that negative and positive values might be worth testing, or distances across the equator/meridian with a changing sign of lat or lon or even using the same for all to make sure the distance is 0.

Reply:
Hi @dargueso,

Some weeks ago, Jonas and I started building a test for this function, see here: https://github.com/matthiasdemuzere/w2w/blob/53aa5c1253f7f132046b2791f557ba806097d867/tests/w2w_test.py#L47

For the test, we just cropped a small ROI out of the LCZ map, and use that to test the routines.

Whilst applying the routine on this small extent, we noticed that some pixels did not change as expected? See notes at lines 73 and 74. I wonder if this is somehow related to the #22 Issue?

Not sure, but I do think it makes sense to have a more in-depth look at this routine?

Reply:
Hi @matthiasdemuzere and @theendlessriver13, 

I guess you mean the routine that removes urban grid points, right?
I can review it and try to find an answer to both your issue and #22. It may be related.

I made some tests for the distance calculation, but I can include a systematic test through pytest as suggested by @theendlessriver13. I'm not very familiar with pytest, but it shouldn't be too hard. 

I'll keep you posted.

Reply:
Hi @matthiasdemuzere, 
I think this behaviour you're noting is expected, unless I reading something wrong. 
That pixel has LU_INDEX 12 and thus it is not changed in the new version.
However, it is composed by 75% of land_cat 12 and 25% of land_cat 13 (urban). In order to remove that percentage, we force it to be 100% land_cat 12. 

Does it make sense? Is there anything I'm missing here?

I'll have a look at issue #22.